lte/cloud/go/services/subscriberdb/servicers/subscriberdb_servicer.go

```diff
@@ -238,6 +266,19 @@ func (s *subscriberdbServicer) getDigestInfo(clientDigest *lte_protos.Digest, ne
 	return digestProto, noUpdates
 }
 
+// shouldCloudDirectedResync returns whether a gateway requires a orc8r-directed resync by checking its
+// last resync time.
+func (l *subscriberdbServicer) shouldCloudDirectedResync(network string, gateway string) bool {
+	lastResyncTime, err := l.lastResyncTimeStore.Get(network, gateway)
+	// If check last resync time in store fails, swallow the error and stick to the original callpath
+	if err != nil {
+		glog.Errorf("check last resync time of gateway %+v of network %+v: %+v", gateway, network, err)
+		return false
+	}
+	shouldResync := uint64(time.Now().Unix())-lastResyncTime > l.resyncIntervalSecs
```

Comment:
We want to deterministically jitter the "actual" resync interval, on a per-gateway basis

This is because we don't want all the AGWs to resync at even around the same time, which would cause a huge spike in CPU usage etc. and likely DoS ourselves

Use https://github.com/magma/magma/pull/7719/files#diff-dd90da7e040d330222abb39f7733b7c9c898b3064e6c3c58e5c98fe55088b651R662 -- make that fn generic enough to reuse here, then make use of it